File: web/test-artifacts/agentic-stress/_screenshot.py
"""Headless screenshot of MACS agentic-hell session showing safety_warning UI."""
import asyncio
import sys
from playwright.async_api import async_playwright

COOKIE_FILE = "/tmp/macs-claude-cookie.txt"
BASE = "http://100.81.47.91:8101"
OUT = "/Users/shaka-mac-mini/coding-projects/macs/web/test-artifacts/agentic-stress/safety-warning-ui.png"


def read_cookie() -> str:
    for ln in open(COOKIE_FILE):
        s = ln.strip()
        if not s:
            continue
        if s.startswith("#HttpOnly_"):
            s = s[len("#HttpOnly_"):]
        elif s.startswith("#"):
            continue
        parts = s.split("\t")
        if len(parts) >= 7 and parts[5] == "pw_session":
            return parts[6]
    raise SystemExit("no pw_session cookie")


async def main():
    cookie = read_cookie()
    async with async_playwright() as pw:
        browser = await pw.chromium.launch(headless=True)
        context = await browser.new_context(viewport={"width": 1440, "height": 900})
        await context.add_cookies([{
            "name": "pw_session",
            "value": cookie,
            "domain": "100.81.47.91",
            "path": "/",
            "httpOnly": True,
            "secure": False,
            "sameSite": "Lax",
        }])
        page = await context.new_page()
        page.on("console", lambda m: print(f"  [console.{m.type}] {m.text[:200]}"))
        page.on("pageerror", lambda e: print(f"  [pageerror] {str(e)[:200]}"))

        print("navigating…")
        await page.goto(BASE + "/", wait_until="networkidle", timeout=30000)
        await page.wait_for_timeout(1000)

        print("clicking agentic-hell…")
        btn = page.locator("text=/agentic.?hell/i").first
        await btn.wait_for(timeout=8000)
        await btn.click()
        await page.wait_for_timeout(1500)

        # Scroll to latest safety_warning
        count = await page.evaluate(
            "() => { const els = document.querySelectorAll('[data-testid^=\"safety-warning-\"]');"
            "if (els.length) els[els.length-1].scrollIntoView({block:'center'}); return els.length; }"
        )
        print(f"safety_warning elements on page: {count}")
        await page.wait_for_timeout(500)

        await page.screenshot(path=OUT, full_page=False)
        print(f"saved → {OUT}")
        await browser.close()
        return count


count = asyncio.run(main())
sys.exit(0 if count > 0 else 2)

File: web/test-artifacts/agentic-stress/_screenshot.mjs
// Headless screenshot of MACS agentic-hell session showing safety_warning UI
import { chromium } from '@playwright/test'
import fs from 'fs'

const COOKIE_FILE = '/tmp/macs-claude-cookie.txt'
const BASE = 'http://100.81.47.91:8101'
const PROJECT_ID = 22  // agentic-hell
const OUT = '/Users/shaka-mac-mini/coding-projects/macs/web/test-artifacts/agentic-stress/safety-warning-ui.png'

// Parse libcurl cookie file → cookie value
function readCookie() {
  for (const ln of fs.readFileSync(COOKIE_FILE, 'utf8').split('\n')) {
    if (!ln.trim()) continue
    let s = ln.trim()
    if (s.startsWith('#HttpOnly_')) s = s.slice('#HttpOnly_'.length)
    else if (s.startsWith('#')) continue
    const parts = s.split('\t')
    if (parts.length >= 7 && parts[5] === 'pw_session') return parts[6]
  }
  throw new Error('no pw_session cookie')
}

const cookie = readCookie()
const browser = await chromium.launch({ headless: true })
const context = await browser.newContext({
  viewport: { width: 1440, height: 900 },
  ignoreHTTPSErrors: true,
})
await context.addCookies([{
  name: 'pw_session',
  value: cookie,
  domain: '100.81.47.91',
  path: '/',
  httpOnly: true,
  secure: false,
  sameSite: 'Lax',
}])
const page = await context.newPage()
page.on('console', (m) => console.log('  [console]', m.type(), m.text().slice(0, 200)))
page.on('pageerror', (e) => console.log('  [pageerror]', e.message.slice(0, 200)))

console.log('navigating…')
await page.goto(BASE + '/', { waitUntil: 'networkidle', timeout: 30000 })
await page.waitForTimeout(800)

// Click agentic-hell project in sidebar
console.log('clicking agentic-hell…')
const projectBtn = page.locator('text=/agentic.?hell/i').first()
await projectBtn.waitFor({ timeout: 8000 })
await projectBtn.click()
await page.waitForTimeout(1200)

// Scroll to the latest safety_warning if any
await page.evaluate(() => {
  const els = document.querySelectorAll('[data-testid^="safety-warning-"]')
  if (els.length) els[els.length - 1].scrollIntoView({ behavior: 'instant', block: 'center' })
})
await page.waitForTimeout(400)

await page.screenshot({ path: OUT, fullPage: false })
console.log('saved →', OUT)
const has = await page.evaluate(() => document.querySelectorAll('[data-testid^="safety-warning-"]').length)
console.log('safety_warning elements on page:', has)
await browser.close()
